### ISSUE TITLE
Function to get size of single glyph in font

### ISSUE THREAD

**Issue by codecat** [memononen/nanovg#397]
It would be nice to have a function such as this:
```c++
void nvgGetGlyphSize(NVGcontext* ctx, const char* pch, float* w, float* h);
```
The given char is a `const char*` here to support iterating over codepoints, but perhaps that can be done better.

My aim for this is to ease stb_textedit implementation.

**Comment by codecat** [comment]
I ended up with a workaround:
```c++
NVGglyphPosition pos[2];
nvgTextGlyphPositions(vg, 0, 0, pch, nullptr, pos, 2);
```
Basically using `pos[1].x` here as the width of a glyph, which would then also contain kerning and space width.

**Comment by memononen** [comment]
I think it this would make interesting case for the demo too.

Here's how caret location from mouse button is handled in the demo:
https://github.com/memononen/nanovg/blob/master/example/demo.c#L887

I wonder if that could be mapped to `STB_TEXTEDIT_LAYOUTROW` and `STB_TEXTEDIT_GETWIDTH` and some caching?

Calling `nvgTextGlyphPositions` incrementally for each character probably does not produce correct results i.e. with kerning.

**Comment by codecat** [comment]
For now I am focusing my current implementation on single-line entry and will move to multi-line entry soon. (I was indeed planning to use `nvgTextBreakLines` for this)

It seems like `nvgTextGlyphPositions` returns acceptable results so far (for short test strings, anyway).

**Comment by memononen** [comment]
I think `stb_textedit` calls `STB_TEXTEDIT_LAYOUTROW` always before it calls `STB_TEXTEDIT_GETWIDTH`. Which means that you could call allocate enough entries in an array of `NVGglyphPosition` and call `nvgTextGlyphPositions` for the whole text. Then `TES_GetWidth` would just return an entry in that array.

So basically you cache a row of glyph positions.

**Comment by codecat** [comment]
You're right, that'd be the best way to go about it! Thanks for the advice! :) 👍
